[Operational Issue]: ValidateInstallers scenario tests fail with NU1301 / 401 Unauthorized on internal feeds (auth not passed into Docker container)

### Error

The `ValidateInstallers_Linux_arm64` leg (step **Validate installer packages**) fails during the SDK scenario tests, which run **inside a Docker container** (`Microsoft.DotNet.Installer.Tests` → `DistroTest` → `docker run`). NuGet restore inside the container cannot authenticate to the internal Azure DevOps feeds and fails with `NU1301` / `401 (Unauthorized)`:

```
SdkTemplateTestsComplex_Console_CSharp.csproj : error NU1301: Unable to load the service index for source
  https://pkgs.dev.azure.com/dnceng/internal/_packaging/darc-int-dotnet-dotnet-f7d90799/nuget/v3/index.json.
SdkTemplateTestsComplex_Console_CSharp.csproj : error NU1301:   Response status code does not indicate success: 401 (Unauthorized).
```

The same `401 (Unauthorized)` occurs for every internal feed referenced by the scenario tests:
- `darc-int-dotnet-dotnet-f7d90799`
- `dotnet10-internal`
- `dotnet10-internal-transport`
- `dotnet-eng-internal`
- `dotnet-tools-internal`

- **Pipeline:** dotnet-unified-build (definition 1330)
- **Branch:** `release/10.0.4xx`
- **Build:** 20260715.18 (buildId 3023455)

### Links

- Failing build (leg `ValidateInstallers_Linux_arm64`, task *Validate installer packages*): https://dev.azure.com/dnceng/internal/_build/results?buildId=3023455&view=logs&j=a6a3a584-660a-5ea0-ec00-3008ec772a89&t=adc9f0dc-a0b8-5f3a-de57-ecb6a3ddbae7
- Similar fix (same 401 internal-feed-auth class, forwarding the NuGet credential-provider environment across the `sudo` boundary for offline validation test restore): https://github.com/dotnet/dotnet/pull/7696

### Root Cause

Following the PAT-less internal feed auth migration (#7686, #7688), the Azure Artifacts NuGet credential provider authenticates using a provisioned session token located via environment (e.g. `HOME` / credential-provider session token env). The SDK installer scenario tests run the restore **inside a Docker container** (`LinuxInstallerTests.DistroTest` builds an image and invokes `DockerHelper.Run` → `docker run`). The container is started without forwarding the credential-provider environment (the `docker run` only mounts the log directory via `-v`), so restore inside the container falls back to unauthenticated and returns `401 (Unauthorized)` against the internal feeds.

This is the Docker-container analog of #7696, which fixed the same failure for the offline validation legs by preserving the credential-provider environment across `sudo` (`--preserve-env=HOME`). The Docker path needs the equivalent env-var passthrough into the container (e.g. `docker run -e ...`).

### Blocking

- [ ] This issue is blocking

### Affected Releases

- [ ] .NET 10
- [x] .NET 10 (SDK band build)
- [ ] .NET 11

--- a/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs
+++ b/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs
@@ -90,6 +90,18 @@ public partial class LinuxInstallerTests : IDisposable
     private const string DotnetApphostPackPrefix = "dotnet-apphost-pack-";
     private const string DotnetSdkPrefix = "dotnet-sdk-";
     private const string DowngradeFxVersionsScript = "downgrade-fx-versions.sh";
+    private const string ContainerNuGetPluginsDir = "/root/.nuget/plugins";
+    private const string ContainerCredentialProviderCacheDir = "/root/.local/share/MicrosoftCredentialProvider";
+    private static readonly string[] NuGetAuthEnvironmentVariables =
+    [
+        "ARTIFACTS_CREDENTIALPROVIDER_EXTERNAL_FEED_ENDPOINTS",
+        "ARTIFACTS_CREDENTIALPROVIDER_FEED_ENDPOINTS",
+        "ARTIFACTS_CREDENTIALPROVIDER_SESSIONTOKENCACHE_ENABLED",
+        "NUGET_CREDENTIALPROVIDER_SESSIONTOKENCACHE_ENABLED",
+        "VSS_NUGET_ACCESSTOKEN",
+        "VSS_NUGET_EXTERNAL_FEED_ENDPOINTS",
+        "VSS_NUGET_URI_PREFIXES"
+    ];
 
     public static bool IncludeRpmTests => Config.TestRpmPackages;
     public static bool IncludeDebTests => Config.TestDebPackages;
@@ -382,8 +394,9 @@ public partial class LinuxInstallerTests : IDisposable
             buildCompleted = true;
 
             // Mount the host log directory to the container
-            string optionalRunArgs = $"-v {hostLogDir}:{containerLogDir}";
-            output = _dockerHelper.Run(tag, tag, testCommand, optionalRunArgs: optionalRunArgs);
+            List<string> optionalRunArgs = [$"-v {QuoteDockerArgument($"{hostLogDir}:{containerLogDir}")}"];
+            optionalRunArgs.AddRange(GetNuGetAuthDockerRunArgs());
+            output = _dockerHelper.Run(tag, tag, testCommand, optionalRunArgs: string.Join(' ', optionalRunArgs));
 
             int testResultsSummaryIndex = output.IndexOf("Tests run: ");
             if (testResultsSummaryIndex >= 0)
@@ -424,6 +437,47 @@ public partial class LinuxInstallerTests : IDisposable
 
         return scenarioTestsBinary.Replace(_contextDir, "").Replace("\\", "/");
     }
+
+    private static IEnumerable<string> GetNuGetAuthDockerRunArgs()
+    {
+        List<string> args = NuGetAuthEnvironmentVariables
+            .Where(envVar => !string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable(envVar)))
+            .Select(envVar => $"-e {envVar}")
+            .ToList();
+
+        if (args.Count == 0)
+        {
+            return args;
+        }
+
+        string? home = Environment.GetEnvironmentVariable("HOME");
+        if (string.IsNullOrWhiteSpace(home))
+        {
+            home = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+        }
+
+        if (string.IsNullOrWhiteSpace(home))
+        {
+            return args;
+        }
+
+        string pluginsDir = Path.Combine(home, ".nuget", "plugins");
+        if (Directory.Exists(pluginsDir))
+        {
+            args.Add($"-v {QuoteDockerArgument($"{pluginsDir}:{ContainerNuGetPluginsDir}:ro")}");
+        }
+
+        string credentialProviderCacheDir = Path.Combine(home, ".local", "share", "MicrosoftCredentialProvider");
+        if (Directory.Exists(credentialProviderCacheDir))
+        {
+            args.Add($"-v {QuoteDockerArgument($"{credentialProviderCacheDir}:{ContainerCredentialProviderCacheDir}")}");
+        }
+
+        return args;
+    }
+
+    private static string QuoteDockerArgument(string value) =>
+        $"\"{value.Replace("\"", "\\\"")}\"";
 
     private List<string> GetPackageList(string baseImage, PackageType packageType)
     {

